### PR TITLE
feat: use bufferline instead of barbar

### LINKED ISSUE #2100: [Feature]: Rename lvim.builtin.bufferline config to lvim.builtin.barbar
### Problem description

I was a bit surprised to learn that LunarVim is using barber instead of bufferline. With the rest of the builtin configuration options, the config value is named after the plugin. However, since barbar is configured using `lvim.builtin.bufferline`, this led me to look at [bufferline.nvim](https://github.com/akinsho/bufferline.nvim)'s config instead.

### LunarVim version

Latest

### Neovim version (>= 0.5)

v0.6.0

### Relevant log output

_No response_

### Screenshots

_No response_

### I have

- [X] Read the readme
- [X] Read the wiki
- [X] Searched for similar issues

### LINKED ISSUE #2221: Serious issue if you create new tmux pane when you are already in tmux zoom mode
### Problem description

1. use default lvim config
2. open tmux, open 2 panes
3. zoom one pane
4. open lvim with a file
5. crate a new tmux pane

Check the lvim pane, the text is a mess, can not be restored, can not do anything inside the lvim, AND the memory for the lvim instance is rapidly increasing. (Check the gif screenshot)

### LunarVim version

2fa176f

### Neovim version (>= 0.6.1)

NVIM v0.6.1

### Operating system/version

Arch Linux

### Relevant log output

```shell
`:checkhealth`


null-ls: require("null-ls.health").check()
========================================================================
  - INFO: no sources registered

nvim: health#nvim#check
========================================================================
## Configuration
  - WARNING: Missing user config file: /home/chz/.config/nvim/init.vim
    - ADVICE:
      - :help |init.vim|

## Performance
  - OK: Build type: RelWithDebInfo

## Remote Plugins
  - OK: Up to date

## terminal
  - INFO: key_backspace (kbs) terminfo entry: key_backspace=^H
  - INFO: key_dc (kdch1) terminfo entry: key_dc=\E[3~
  - INFO: $TERM_PROGRAM='tmux'
  - INFO: $COLORTERM='truecolor'

## tmux
  - OK: escape-time: 50
  - INFO: Checking stuff
  - OK: focus-events: on
  - INFO: $TERM: screen-256color

nvim-lsp-installer: require("nvim-lsp-installer.health").check()
========================================================================
## nvim-lsp-installer report
  - OK: neovim version >= 0.6.0
  - WARNING: **Composer**: not available
  - WARNING: **PHP**: not available
  - WARNING: **javac**: not available
  - WARNING: **julia**: not available
  - OK: **tar**: `tar (GNU tar) 1.34`
  - OK: **gzip**: `gzip 1.11`
  - OK: **curl**: `curl 7.81.0 (x86_64-pc-linux-gnu) libcurl/7.81.0 OpenSSL/1.1.1m zlib/1.2.11 brotli/1.0.9 zstd/1.5.2 libidn2/2.3.2 libpsl/0.21.1 (+libidn2/2.3.0) libssh2/1.10.0 nghttp2/1.46.0`
  - OK: **wget**: `GNU Wget 1.21.2 built on linux-gnu.`
  - OK: **python3**: `Python 3.10.2`
  - OK: **node**: `v17.3.0`
  - OK: **Ruby**: `ruby 3.0.3p157 (2021-11-24 revision 3fb7d2cadc) [x86_64-linux]`
  - OK: **Go**: `go version go1.17.6 linux/amd64`
  - OK: **bash**: `GNU bash, version 5.1.16(1)-release (x86_64-pc-linux-gnu)`
  - OK: **sh**: `Ok`
  - OK: **java**: `Ok`
  - OK: **RubyGem**: `3.3.5`
  - OK: **npm**: `8.3.2`
  - OK: **pip3**: `pip 20.3.4 from /usr/lib/python3.10/site-packages/pip (python 3.10)`

nvim-treesitter: require("nvim-treesitter.health").check()
========================================================================
## Installation
  - OK: `tree-sitter` found 0.20.1 (062421dece3315bd6f228ad6d468cba083d0a2d5) (parser generator, only needed for :TSInstallFromGrammar)
  - OK: `node` found v17.3.0 (only needed for :TSInstallFromGrammar)
  - OK: `git` executable found.
  - OK: `cc` executable found. Selected from { vim.NIL, "cc", "gcc", "clang", "cl", "zig" }
    Version: cc (GCC) 11.1.0
  - OK: Neovim was compiled with tree-sitter runtime ABI version 13 (required >=13). Parsers must be compatible with runtime ABI.

## Parser/Features H L F I J
  - yaml           ✓ ✓ ✓ ✓ ✓ 
  - json           ✓ ✓ ✓ ✓ . 
  - cmake          ✓ . ✓ . . 
  - python         ✓ ✓ ✓ ✓ ✓ 
  - toml           ✓ ✓ ✓ ✓ ✓ 
  - markdown       ✓ . ✓ . ✓ 
  - c              ✓ ✓ ✓ ✓ ✓ 
  - typescript     ✓ ✓ ✓ ✓ ✓ 
  - css            ✓ . ✓ ✓ ✓ 
  - ninja          ✓ . ✓ ✓ . 
  - vim            ✓ ✓ . . ✓ 
  - make           ✓ . . . ✓ 
  - java           x ✓ . x x 
  - fish           ✓ ✓ ✓ ✓ ✓ 
  - bash           ✓ ✓ ✓ . ✓ 
  - lua            x x x x x 
  - javascript     ✓ ✓ ✓ ✓ ✓ 
  - cpp            x x ✓ ✓ ✓ 
  - rust           ✓ ✓ ✓ ✓ ✓ 

  Legend: H[ighlight], L[ocals], F[olds], I[ndents], In[j]ections
         +) multiple parsers found, only one will be used
         x) errors found in the query, try to run :TSUpdate {lang}

## The following errors have been detected:
  - ERROR: java(highlights): ...rtgX/usr/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 2073
  - ERROR: java(indents): ...rtgX/usr/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 239
  - ERROR: java(injections): ...rtgX/usr/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 2
  - ERROR: lua(highlights): ...rtgX/usr/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 503
  - ERROR: lua(locals): ...rtgX/usr/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 15
  - ERROR: lua(folds): ...rtgX/usr/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 92
  - ERROR: lua(indents): ...rtgX/usr/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 29
  - ERROR: lua(injections): ...rtgX/usr/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid field at position 18
  - ERROR: cpp(highlights): ...rtgX/usr/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 3277
  - ERROR: cpp(locals): ...rtgX/usr/share/nvim/runtime/lua/vim/treesitter/query.lua:161: query: invalid node type at position 1938

provider: health#provider#check
========================================================================
## Clipboard (optional)
  - OK: Clipboard tool found: xclip

## Python 2 provider (optional)
  - WARNING: No Python executable found that can `import neovim`. Using the first available executable for diagnostics.
  - ERROR: Python provider error:
    - ADVICE:
      - provider/pythonx: Could not load Python 2:
          python2 not found in search path or not executable.
          python2.7 not found in search path or not executable.
          python2.6 not found in search path or not executable.
          /bin/python is Python 3.10 and cannot provide Python 2.
  - INFO: Executable: Not found

## Python 3 provider (optional)
  - INFO: `g:python3_host_prog` is not set.  Searching for python3 in the environment.
  - INFO: Multiple python3 executables found.  Set `g:python3_host_prog` to avoid surprises.
  - INFO: Executable: /bin/python3
  - INFO: Other python executable: /sbin/python3
  - INFO: Other python executable: /usr/bin/python3
  - INFO: Other python executable: /usr/sbin/python3
  - INFO: Python version: 3.10.2
  - INFO: pynvim version: 0.4.3
  - OK: Latest pynvim is installed.

## Python virtualenv
  - OK: no $VIRTUAL_ENV

## Ruby provider (optional)
  - INFO: Ruby: ruby 3.0.3p157 (2021-11-24 revision 3fb7d2cadc) [x86_64-linux]
  - WARNING: `neovim-ruby-host` not found.
    - ADVICE:
      - Run `gem install neovim` to ensure the neovim RubyGem is installed.
      - Run `gem environment` to ensure the gem bin directory is in $PATH.
      - If you are using rvm/rbenv/chruby, try "rehashing".
      - See :help |g:ruby_host_prog| for non-standard gem installations.

## Node.js provider (optional)
  - INFO: Node.js: v17.3.0
  - INFO: Nvim node.js host: /home/chz/.npms/lib/node_modules/neovim/bin/cli.js
  - OK: Latest "neovim" npm/yarn package is installed: 4.10.1

## Perl provider (optional)
  - ERROR: perl provider error:
    - ADVICE:
      - "Neovim::Ext" cpan module is not installed

telescope: require("telescope.health").check()
========================================================================
## Checking for required plugins
  - OK: plenary installed.
  - OK: nvim-treesitter installed.

## Checking external dependencies
  - OK: rg: found ripgrep 12.1.1 (rev 7cb211378a)
  - OK: fd: found fd 8.3.1

## ===== Installed extensions =====

## Telescope Extension: `fzf`
  - INFO: No healthcheck provided

## Telescope Extension: `projects`
  - INFO: No healthcheck provided

vim.lsp: require("vim.lsp.health").check()
========================================================================
  - INFO: LSP log level : WARN
  - INFO: Log path: /home/chz/.cache/nvim/lsp.log
  - INFO: Log size: 475 KB

vim.treesitter: require("vim.treesitter.health").check()
========================================================================
  - INFO: Runtime ABI version : 13
  - OK: Loaded parser for bash: ABI version 13
  - OK: Loaded parser for cmake: ABI version 13
  - OK: Loaded parser for cpp: ABI version 13
  - OK: Loaded parser for css: ABI version 13
  - OK: Loaded parser for fish: ABI version 13
  - OK: Loaded parser for java: ABI version 13
  - OK: Loaded parser for javascript: ABI version 13
  - OK: Loaded parser for json: ABI version 13
  - OK: Loaded parser for lua: ABI version 13
  - OK: Loaded parser for make: ABI version 13
  - OK: Loaded parser for markdown: ABI version 13
  - OK: Loaded parser for ninja: ABI version 13
  - OK: Loaded parser for python: ABI version 13
  - OK: Loaded parser for rust: ABI version 13
  - OK: Loaded parser for toml: ABI version 13
  - OK: Loaded parser for typescript: ABI version 13
  - OK: Loaded parser for vim: ABI version 13
  - OK: Loaded parser for yaml: ABI version 13
  - OK: Loaded parser for c: ABI version 13

which_key: health#which_key#check
========================================================================
## WhichKey: checking conflicting keymaps
```

### Screenshots
Peek 2022-01-30 01-44

### PR DESCRIPTION
**Pull request by abzcoding**
# Description

Use [bufferline](https://github.com/akinsho/bufferline.nvim) instead of [barbar](https://github.com/romgrk/barbar.nvim)

Fixes #2221 
Fixes #2100

### CODE REVIEW

File: lua/lvim/core/bufferline.lua [hunk 1]
```diff
@@ -1,18 +1,127 @@
...
+      show_buffer_icons = true, -- disable filetype icons for buffers
+      show_buffer_close_icons = true,
+      show_close_icon = false,
+      show_tab_indicators = true,
+      persist_buffer_sort = true, -- whether or not custom sorted buffers should persist
+      -- can also be a table containing 2 custom separators
+      -- [focused and unfocused]. eg: { '|', '|' }
+      separator_style = "thin",
+      enforce_regular_tabs = false,
+      always_show_bufferline = false,
```
Comment (kylo252) [inline lua/lvim/core/bufferline.lua:116]: This one might be a bit confusing, I even forgot about it. I think it controls whether a single buffer should still show the bufferline. If it's set to `false`, then opening a buffer from Alpha won't show the bufferline.

File: lua/lvim/core/bufferline.lua [hunk 2]
```diff
@@ -1,18 +1,127 @@
...
+end
+
 M.config = function()
   lvim.builtin.bufferline = {
     active = true,
     on_config_done = nil,
     keymap = {
       normal_mode = {},
     },
+    options = {
```
Comment (kylo252) [inline lua/lvim/core/bufferline.lua:43]:
```suggestion
    highlights = {
      background = {
        gui = "italic",
      },
      buffer_selected = {
        gui = "bold",
      },
    },
    options = {
```
This is a bit of a personal perference (I find the defult confusing 🥲 ), but we should at least support overriding these options.

File: lua/lvim/core/bufferline.lua [hunk 3]
```diff
@@ -1,18 +1,127 @@
...
+      always_show_bufferline = false,
+      sort_by = "id",
+    },
   }
 end
 
 M.setup = function()
   require("lvim.keymappings").load(lvim.builtin.bufferline.keymap)
-
+  require("bufferline").setup { options = lvim.builtin.bufferline.options }
```
Comment (kylo252) [inline lua/lvim/core/bufferline.lua:124]:
```suggestion
  require("bufferline").setup {
    options = lvim.builtin.bufferline.options,
    highlights = lvim.builtin.bufferline.highlights,
  }

```

File: lua/lvim/core/bufferline.lua (hunk 1 above)
Comment (abzcoding) [reply lua/lvim/core/bufferline.lua:124]: yes which is what i intended to do and what i think was missing from barbar

**Review by kylo252 — APPROVED** [review]
LGTM! 🚀

File: lua/lvim/core/bufferline.lua (hunk 1 above)
Comment (kylo252) [reply lua/lvim/core/bufferline.lua:124]: Sounds good, and people can always change it.

--- 

NB: Please edit the description to include https://github.com/LunarVim/LunarVim/issues/2100 as well :)

**Pull request merged**